app/src/test/java/com/example/movieapp/ui/main/viewmodel/HomeViewModelTest.java

```diff
@@ -0,0 +1,172 @@
+package com.example.movieapp.ui.main.viewmodel;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import com.example.movieapp.data.api.NetworkStatus;
+import com.example.movieapp.data.api.Resource;
+import com.example.movieapp.data.model.MovieData;
+import com.example.movieapp.data.model.PopularMoviesData;
+import com.example.movieapp.data.model.PopularTvShowsData;
+import com.example.movieapp.data.model.TopRatedMoviesData;
+import com.example.movieapp.data.model.TopRatedTvShowsData;
+import com.example.movieapp.data.model.UpComingMoviesData;
+import com.example.movieapp.data.repository.home.HomeRepository;
+
+import org.junit.After;
+import org.junit.Before;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HomeViewModelTest {
+    @Rule
+    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+private HomeViewModel homeViewModel;
+@Mock
+private HomeRepository homeRepository;
+    @Before
+    public void setUp() throws Exception {
+        homeViewModel= new HomeViewModel();
+        homeViewModel.homeRepository=homeRepository;
+        MutableLiveData<Resource<PopularMoviesData>> popularMoviesData = getPopularMoviesDataTest();
+        MutableLiveData<Resource<TopRatedMoviesData>> topRatedMoviesData = getTopRatedMoviesDataTest();
+        MutableLiveData<Resource<PopularTvShowsData>> popularTvShowsData = getPopularTvShowsDataTest();
+        MutableLiveData<Resource<TopRatedTvShowsData>> topRatedTvShowsData = getTopRatedTvShowsDataTest();
+        MutableLiveData<Resource<UpComingMoviesData>> upComingMoviesData = getUpComingMoviesDataTest();
+        when(homeRepository.getPopularMoviesData()).thenReturn(popularMoviesData);
+        when(homeRepository.getTopRatedMoviesData()).thenReturn(topRatedMoviesData);
+        when(homeRepository.getPopularTvShowsData()).thenReturn(popularTvShowsData);
+        when(homeRepository.getTopRatedTvShowsData()).thenReturn(topRatedTvShowsData);
+        when(homeRepository.getUpComingMoviesData()).thenReturn(upComingMoviesData);
+    }
+
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void getPopularMovies() {
+        LiveData<List<MovieData>> res = homeViewModel.getPopularMovies();
+         verify(homeRepository).getPopularMoviesData();
+        Assert.assertEquals(getMovieDataListTest().get(0).getId(), res.getValue().get(0).getId());
+    }
+
+    @Test
```

Comment:
Add comments before each test case, so it's easy to understand what the test case is actually doing